src/figuresClasses.ts

```diff
@@ -1,19 +1,89 @@
-export interface Figure {
+enum Color {
+  'red', 'green', 'blue'
+}
 
+export interface Figure {
+  shape: 'triangle' | 'circle' |'rectangle';
+  color: Color;
+  getArea: Function;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: 'triangle',
+  ) {
+    if ((this.a <= 0 || this.b <= 0 || this.c <= 0)
+        || this.c >= (this.a + this.b) || this.b >= (this.a + this.c)
+        || this.a >= (this.b + this.c)) {
+      throw new Error('ERROR');
+    }
 
-}
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
```

Comment:
In TypeScript, when you declare a constructor parameter with an access modifier TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property.
So in your code, public a: number in the constructor is equivalent to writing this.a = a. You can safely remove this.